packages/cojson-transport-ws/src/index.ts

```diff
@@ -40,6 +40,24 @@ const g: typeof globalThis & {
     }[];
 } = globalThis;
 
+function promiseWithResolvers<R>() {
+    let resolve = (_: R) => {};
+    let reject = (_: unknown) => {};
+  
+    const promise = new Promise<R>((_resolve, _reject) => {
+      resolve = _resolve;
+      reject = _reject;
+    });
+  
+    return {
+      promise,
+      resolve,
+      reject,
+    };
+  }
+
+const BUFFER_LIMIT = 100_000;
```

Comment:
Using a smaller buffer limit to pause the queue processing earlier.

This way we have more "controlled" buffer and leave to the websocket control only a small amount of messages.